### PR TITLE
n2n communication

### PR DESCRIPTION
**Pull request by bartvanb**
First functional version where algorithms can connect to one another

### CODE REVIEW

File: dockerfiles/vpn-client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+RUN apk add --no-cache \
+        bind-tools \
+        openvpn
+
+RUN mkdir /app
+
+COPY . /app/
+
+ENTRYPOINT ["/app/entry.sh"]
```
Comment (frankcorneliusmartin) [inline dockerfiles/vpn-client/Dockerfile]: This Dockerfile should be moved into the master repository.

File: dockerfiles/vpn-client/entry.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    --pull-filter ignore "route-ipv6" \
+    --pull-filter ignore "ifconfig-ipv6" \
+    --script-security 2 \
+    --up-restart \
+    --ping-exit 10 \
+    --cd /mnt/vpn &
+openvpn_child=$!
+
+wait $openvpn_child
+echo "Exiting, status $openvpn_child"
```
Comment (frankcorneliusmartin) [inline dockerfiles/vpn-client/entry.sh]: The we would have to move this file also

File: dockerfiles/vpn-configurator/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+FROM alpine
+
+RUN apk add iproute2
```
Comment (frankcorneliusmartin) [inline dockerfiles/vpn-configurator/Dockerfile]: Aaand this one

File: vantage6/node/__init__.py [hunk 4]
```diff
@@ -171,25 +187,30 @@ def __init__(self, ctx):
 
         # setup docker isolated network manager
         isolated_network_mgr = \
-            IsolatedNetworkManager(f"{ctx.docker_network_name}-net")
+            IsolatedNetworkManager(f"{self.ctx.docker_network_name}-net")
```
Comment (frankcorneliusmartin) [inline vantage6/node/__init__.py]: Why do we append the  `-net` or why is that not included in the `ctx.docker_network_name`?

File: vantage6/node/globals.py [hunk 5]
```diff
@@ -30,10 +26,8 @@
 #    VPN CONFIGURATION RELATED CONSTANTS
 #
 # TODO move part of these constants elsewhere?! Or make context?
-VPN_CLIENT_IMAGE = 'algorithm-container-network_openvpn-client'
-NETWORK_CONFIG_IMAGE = 'network-config'
-VPN_SUBNET = '10.76.0.0/16'
-VPN_CONFIG_FILE = 'data/vpn-config.ovpn.conf'
-LOCAL_SUBNET_START = '172.'   # start of local IP addresses
+VPN_CLIENT_IMAGE = 'harbor2.vantage6.ai/infrastructure/vpn-client'
+NETWORK_CONFIG_IMAGE = 'harbor2.vantage6.ai/infrastructure/vpn-configurator'
```
Comment (frankcorneliusmartin) [inline vantage6/node/globals.py:30]: Remark: 
These need to be overwritable from the configuration file. I created a separate issue for this: https://github.com/IKNL/vantage6-node/issues/45

Lets do that in a patch or minor version

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]

File: vantage6/node/__init__.py (hunk 4 above)
Comment (bartvanb) [reply vantage6/node/__init__.py]: Good question. This is the only place in all repo's we use this property, so I added '-net' to this property in the Context class and removed the -net here.

File: dockerfiles/vpn-client/Dockerfile (hunk 1 above)
Comment (bartvanb) [reply dockerfiles/vpn-client/Dockerfile]: done

File: dockerfiles/vpn-client/entry.sh (hunk 2 above)
Comment (bartvanb) [reply dockerfiles/vpn-client/entry.sh]: done

File: dockerfiles/vpn-configurator/Dockerfile (hunk 3 above)
Comment (bartvanb) [reply dockerfiles/vpn-configurator/Dockerfile]: done

**Review by frankcorneliusmartin — APPROVED** [review]

**Pull request merged**